contracts/vaults/Base.sol

```diff
@@ -289,15 +297,18 @@ contract  Base is OwnableUpgradeable, ReentrancyGuardUpgradeable {
         pool.lastProcessedTotalAllocPoint = lastPoolUpdate;
     }
 
-    // Safe rewardToken transfer function, transfer HATs from the contract only if they are earmarked for rewards
+    /**
+    * @dev Safe HAT transfer function, transfer rewards from the contract only if there are enough
+    * rewards available.
+    * @param _to The address to transfer the reward to
+    * @param _amount The amount of rewards to transfer
+    * @param _pid The pool id
+   */
     function safeTransferReward(address _to, uint256 _amount, uint256 _pid) internal {
-        if (_amount > rewardAvailable) { 
-            _amount = rewardAvailable; 
-        }
-        rewardAvailable -= _amount;
+        require(rewardAvailable >= _amount, "HVE46");
         rewardToken.transfer(_to, _amount);
-        // TODO: fix return of the requested amount
-        emit SafeTransferReward(_to, _pid, _amount, _amount);
+        rewardAvailable -= _amount;
```

Comment:
but why did you change the order here?
So in general there is a "best practice" rule to do state changes before external calls, to avoid reentrancy attacks. (In this case, the transfer poses no risk, and anyway we call safetransfer in other contexts, but it is still a good pattern to follow)